src/clj/puppetlabs/http/client/sync.clj

```diff
@@ -23,43 +23,63 @@
   (select-keys opts [:url :method :headers :body :decompress-body :as :query-params]))
 
 (defn request-with-client
-  [req client]
-  (let [{:keys [error] :as resp} @(async/request-with-client req nil client)]
+  [req client metric-registry]
+  (let [{:keys [error] :as resp} @(async/request-with-client req nil client metric-registry)]
     (if error
       (throw error)
       resp)))
 
+(defn get-client-metrics
+  ([metric-registry]
+   (async/get-client-metrics metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics metric-registry metric-filter)))
+
+(defn get-client-metrics-data
+  ([metric-registry]
+   (async/get-client-metrics-data metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics-data metric-registry metric-filter)))
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
 (defn request
   [req]
   (with-open [client (async/create-default-client (extract-client-opts req))]
-    (request-with-client (extract-request-opts req) client)))
+    (request-with-client (extract-request-opts req) client nil)))
```

Comment:
I guess this might be a good/useful thing. But I wonder, if someone is just making one-off requests, how much of a use-case is there for tracking metrics on them? If a user really wanted to be tracking metrics on a request, is it that hard for them to create a client and start using that? I don't have a super strong opinion about this, just pondering.